bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -0,0 +1,375 @@
+                    return;
+                }
+            }
+
+        } else {
+            logger.warn("handleMode() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private void handleFunction(Command command) {
+        logger.debug("handleFunction() (command={})", command);
+        if (command instanceof StringType) {
+            Where w = deviceWhere;
+            if (w != null) {
+                try {
+                    FUNCTION function = FUNCTION.valueOf(((StringType) command).toString());
+                    logger.debug("handleFunction() mode={}", function.toString());
+                    send(Thermoregulation.requestWriteFunction(w.value(), function));
+                } catch (OWNException e) {
+                    logger.warn("handleFunction() {}", e.getMessage());
+                } catch (IllegalArgumentException e) {
+                    logger.warn("handleFunction() Unsupported command {} for thing {}", command, getThing().getUID());
+                    return;
+                }
+            }
+        } else {
+            logger.warn("handleFunction() Unsupported command {} for thing {}", command, getThing().getUID());
+        }
+    }
+
+    private QuantityType<Temperature> commandToQuantityType(Command command, Unit<Temperature> unit) {
+        return new QuantityType<Temperature>(command.toFullString());
+    }
+
+    @Override
+    protected void handleMessage(BaseOpenMessage msg) {
+        super.handleMessage(msg);
+        if (msg.isCommand()) {
+            updateModeAndFunction((Thermoregulation) msg);
+        } else {
+            if (msg.getDim() == null) {
+                return;
+            }
+            if (msg.getDim() == Thermoregulation.DIM.TEMPERATURE
+                    || msg.getDim() == Thermoregulation.DIM.PROBE_TEMPERATURE) {
+                updateTemperature((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.TEMP_SETPOINT
+                    || msg.getDim() == Thermoregulation.DIM.COMPLETE_PROBE_STATUS) {
+                updateSetpoint((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.VALVES_STATUS) {
+                updateValveStatus((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.ACTUATOR_STATUS) {
+                updateActuatorStatus((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.FAN_COIL_SPEED) {
+                updateFanCoilSpeed((Thermoregulation) msg);
+            } else {
+                logger.debug("handleMessage() Ignoring unsupported DIM {} for thing {}. Frame={}", msg.getDim(),
+                        getThing().getUID(), msg);
+            }
+        }
+    }
+
+    private void updateModeAndFunction(Thermoregulation tmsg) {
+        logger.debug("updateModeAndFunction() for thing: {} msg={}", thing.getUID(), tmsg);
```

Comment:
Please use the debugger over log messages. Please check all.

Reply:
Got it. I will check them all.